script.js

```diff
@@ -1,4 +1,282 @@
+    playersArray[playerNum - 1].diceNumRolled.join("")
+  );
+  playersArray[playerNum - 1].score += playersArray[playerNum - 1].numResult;
+
+  let msg = `Hi Player ${playerNum}, you rolled ${
+    playersArray[playerNum - 1].diceNumRolled
+  } for your dice.`;
+
+  if (playerNum == totalNumOfPlayers) {
+    let winningNum = Math.max(
+      ...playersArray.map((player) => player.numResult)
+    );
+    let winningPlayer =
+      playersArray.map((player) => player.numResult).indexOf(winningNum) + 1;
+
+    msg =
+      msg +
+      `<br><br>The winner is Player ${winningPlayer} with a value ${winningNum}.<br><br><h3>Leadersboard</h3>${leadersboardMsg(
+        playersArray
+      )}`;
+
+    playerNum = 1;
+    return msg;
+  }
+  playerNum += 1;
+  return msg;
+};
+
+//creating function to run lowest scoring game mode
+const lowestMode = (playersArray, input) => {
+  if (input != "") {
+    return `Hi Player ${playerNum}. Please kindly click the Submit button to roll the dice.`;
+  }
+  for (i = 0; i < numOfDice; i += 1) {
+    playersArray[playerNum - 1].diceNumRolled[i] = rollDice();
+  }
+  playersArray[playerNum - 1].diceNumRolled.sort((a, b) => a - b);
+  playersArray[playerNum - 1].numResult = Number(
+    playersArray[playerNum - 1].diceNumRolled.join("")
+  );
+  playersArray[playerNum - 1].score += playersArray[playerNum - 1].numResult;
+
+  let msg = `Hi Player ${playerNum}, you rolled ${
+    playersArray[playerNum - 1].diceNumRolled
+  } for your dice.`;
+
+  if (playerNum == totalNumOfPlayers) {
+    let winningNum = Math.min(
+      ...playersArray.map((player) => player.numResult)
+    );
+    let winningPlayer =
+      playersArray.map((player) => player.numResult).indexOf(winningNum) + 1;
+
+    msg =
+      msg +
+      `<br><br>The winner is Player ${winningPlayer} with a value ${winningNum}.<br><br><h3>Leadersboard</h3>${leadersboardMsg(
+        playersArray
+      )}`;
+
+    playerNum = 1;
+    return msg;
+  }
+  playerNum += 1;
+  return msg;
+};
+
+//creating function to run knockout mode
+//1. need a separate array of selected players for face off
+//2. the loser in the face off will be removed from the original array
+//3. repeat 1 & 2 until there is only 1 player in the original array and declare winner.
+const knockoutMode = (playersArray, input) => {
+  if (faceOffArray.length == 0) {
+    if (input != "") {
+      return `Hi user. Please kindly click the Submit button to draw the players for face off.`;
+    }
+    let playerA_Index = Math.floor(Math.random() * playersArray.length);
+    let playerB_Index = Math.floor(Math.random() * playersArray.length);
+    while (playerA_Index == playerB_Index) {
+      playerB_Index = Math.floor(Math.random() * playersArray.length);
+    }
+    let playerA = playersArray[playerA_Index];
+    let playerB = playersArray[playerB_Index];
+    faceOffArray.push(playerA);
+    faceOffArray.push(playerB);
+    console.log(faceOffArray);
```

Comment:
it is good to use console log during development but remove them during code submission for clarity and readability of code